File: Priprema K3 - Predavanje/IN/Python/vagon.py
'''
Created on Dec 25, 2018

@author: Marko
'''

class Vagon:

    def __init__(self):
        self.opisTereta = "PRAZAN"
        self.teret = []
        
    def getTezinaRobe(self):
        suma = 0;
        for t in self.teret:
            suma += t.tezinaKG
        return suma
    
    def dodajRobu(self, roba):
        self.teret.append(roba)
        self.opisTereta = self.teret[0].marka
        for t in self.teret:
            if t.marka != self.opisTereta:
                self.opisTereta = "RAZNO"
                break
    
    def __str__(self):
        ret = "teret "
        count = len(self.teret)
        if count == 0:
            ret += "prazan"
        for t in range(count):
            ret += str(self.teret[t])
            if t < count - 1:
                ret += ", "
        ret += " > opisTereta = \"" + self.opisTereta + "\""
        return ret

File: Priprema K3 - Predavanje/IN/Java/Vagon.java
import java.util.ArrayList;

public class Vagon {
	private String opisTereta = "PRAZAN";
	private ArrayList<Roba> teret = new ArrayList<>();
	
	public ArrayList<Roba> getTeret() {
		return teret;
	}

	public int getTezinaRobe() {
		int suma = 0;
		for(int i = 0; i < teret.size(); ++i) {
			suma += teret.get(i).getTezinaKG();
		}
		return suma;
	}
	
	public void dodajRobu(Roba roba) {
		teret.add(roba);
		opisTereta = teret.get(0).getMarka();
		for(int i = 0; i < teret.size(); ++i) {
			if (teret.get(i).getMarka() != opisTereta) {
				opisTereta = "RAZNO";
				break;
			}
		}
	}
	
	@Override
	public String toString() {
		String ret = "teret ";
		int count = teret.size();
		
		if (count == 0)
            ret += "prazan";
		for(int i = 0; i < teret.size(); ++i) {
			ret += teret.get(i);
			if (i < count - 1)
				ret += ", ";
		}
		ret += " > opisTereta = \"" + opisTereta + "\"";

		return ret;
	}
}